### PR TITLE
Upgrade Ruby and gems

### PR DESCRIPTION
**Pull request by andistuder**
Maintenance of Ruby 2.0.0 ends after 24th Feb 2016 [ref](https://www.ruby-lang.org/en/news/2015/12/16/ruby-2-0-0-p648-released/). 

This pull request: 
- upgrades Ruby to 2.3.0
- upgrades Rails to 4.1.14.1
- updates gems where possible to get tests suite to build&pass - see [commit message](https://github.com/andistuder/projectmonitor/commit/ea26876d91dd53edd9451d561a360bcb89932552) for details
- fixes bug where empty array instead of empty string passed to `URI.decode_www_form`

### CODE REVIEW

File: app/models/travis_pro_project.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def self.project_specific_attributes
   # Add ?token= or &token= to the feed_url, as appropriate
   def feed_url
     URI.parse(super).tap do |uri|
-      params = URI.decode_www_form(uri.query || []) << ['token', ci_auth_token]
+      params = URI.decode_www_form(uri.query || '') << ['token', ci_auth_token]
```
Comment (0mega) [inline app/models/travis_pro_project.rb]: Should we have a spec for this?

**Comment by andistuder** [comment]
FYI: I've tried upgrading to the latest Rails (4.2.5.1). But this caused a number of tests to fail. As the focus of this PR is to address the due Ruby upgrade, I decided to pin to the last patch on Rails 4.1.

File: app/models/travis_pro_project.rb (hunk 1 above)
Comment (andistuder) [inline app/models/travis_pro_project.rb]: [travis_pro_project_spec.rb#L9] (https://github.com/pivotal/projectmonitor/blob/master/spec/models/travis_pro_project_spec.rb#L9) covers this: ie. it failed after upgrading the Gems.

**Comment by 0mega** [comment]
LGTM

**Comment by andistuder** [comment]
@odlp @Pinwheeler @dkaplan-pivotal please review asap - ruby 2.0 is no longer maintained.

**Comment by odlp** [comment]
Thanks @andistuder for this work, apologies for the delay in merging.

**Pull request merged**